jquery.flot.hover.js

```diff
@@ -0,0 +1,303 @@
+    function triggerClickHoverEvent(eventname, event, seriesFilter, searchDistance) {
+        var options = plot.getOptions(),
+            offset = plot.offset(),
+            canvasX = event.pageX - offset.left,
+            canvasY = event.pageY - offset.top,
+            pos = plot.c2p({
+                left: canvasX,
+                top: canvasY
+            }),
+            distance = searchDistance !== undefined ? searchDistance : options.grid.mouseActiveRadius;
+
+        pos.pageX = event.pageX;
+        pos.pageY = event.pageY;
+
+        var item = plot.findNearbyItem(canvasX, canvasY, seriesFilter, distance);
+
+        if (item) {
+            // fill in mouse pos for any listeners out there
+            item.pageX = parseInt(item.series.xaxis.p2c(item.datapoint[0]) + offset.left, 10);
+            item.pageY = parseInt(item.series.yaxis.p2c(item.datapoint[1]) + offset.top, 10);
+        }
+
+        if (options.grid.autoHighlight) {
+            // clear auto-highlights
+            for (var i = 0; i < highlights.length; ++i) {
+                var h = highlights[i];
+                if ((h.auto === eventname &&
+                    !(item && h.series === item.series &&
+                        h.point[0] === item.datapoint[0] &&
+                        h.point[1] === item.datapoint[1])) || !item) {
+                    unhighlight(h.series, h.point);
+                }
+            }
+
+            if (item) {
+                highlight(item.series, item.datapoint, eventname);
+            }
+        }
+
+        plot.getPlaceholder().trigger(eventname, [pos, item]);
+    }
+
+    function highlight(s, point, auto) {
+        if (typeof s === "number") {
+            s = series[s];
+        }
+
+        if (typeof point === "number") {
+            var ps = s.datapoints.pointsize;
+            point = s.datapoints.points.slice(ps * point, ps * (point + 1));
+        }
+
+        var i = indexOfHighlight(s, point);
+        if (i === -1) {
+            highlights.push({
+                series: s,
+                point: point,
+                auto: auto
+            });
+
+            plot.triggerRedrawOverlay();
+        } else if (!auto) {
+            highlights[i].auto = false;
+        }
+    }
+
+    function unhighlight(s, point) {
+        if (s == null && point == null) {
+            highlights = [];
+            plot.triggerRedrawOverlay();
+            return;
+        }
+
+        if (typeof s === "number") {
+            s = series[s];
+        }
+
+        if (typeof point === "number") {
+            var ps = s.datapoints.pointsize;
+            point = s.datapoints.points.slice(ps * point, ps * (point + 1));
+        }
+
+        var i = indexOfHighlight(s, point);
+        if (i !== -1) {
+            highlights.splice(i, 1);
+
+            plot.triggerRedrawOverlay();
+        }
+    }
+
+    function indexOfHighlight(s, p) {
+        for (var i = 0; i < highlights.length; ++i) {
```

Comment:
Is there a reason to skip the first element? Maybe a **.filter** would make this cleaner :-)

Reply:
This was a copy-paste from flot :)

Reply:
It doesn't look like :-P